TRACING.md

```diff
@@ -0,0 +1,272 @@
+# Agent Tracing
+
+* [Agent Tracing](#agent-tracing)
+* [Introduction](#introduction)
+* [OpenTracing summary](#opentracing-summary)
+* [Jaeger tracing architecture](#jaeger-tracing-architecture)
+* [Guest to Host communication using VSOCK](#guest-to-host-communication-using-vsock)
+* [Enabling tracing](#enabling-tracing)
+  * [Tracing options](#tracing-options)
+* [Running the agent with tracing enabled](#running-the-agent-with-tracing-enabled)
+* [Requirements](#requirements)
+  * [Host environment](#host-environment)
+  * [Guest environment](#guest-environment)
+* [Limitations](#limitations)
+* [Appendix - Setting up a tracing development environment](#appendix---setting-up-a-tracing-development-environment)
+  * [Set up a debug console](#set-up-a-debug-console)
+  * [Guest journal redirector](#guest-journal-redirector)
+  * [Enable full agent debug](#enable-full-agent-debug)
+
+# Introduction
+
+This document explains agent tracing, which uses 
+the [Jaeger](https://jaegertracing.io) implementation of the [OpenTracing](http://opentracing.io) API.
```

Comment:
@jodh-intel one option would be to decouple the exporting from trace gathering. If we used open census we can export to any backend

https://opencensus.io/exporters/supported-exporters/go/

This will allow Kata to directly export traces to Cloud backends like Azure Monitor, AWS X-Ray (Tracing) and Stackdriver and Datadog.

This is a easy swap out.

Also open tracing has built in support for gRPC tracing

https://github.com/census-instrumentation/opencensus-go/blob/master/examples/grpc/helloworld_server/main.go#L62

Reply:
It does sound like we have an embarrassment of riches when it comes to tracing options. 

I'll have to look at this next week. We already have otgrpc gRPC tracing support in the agent (today) but if you look at this PR, you'll see that I use that for "collated trace", but a custom gRPC tracer for "isolated tracing". So we could potentially switch the collated trace to use opencensus but that isn't "enough" for isolated tracing which traces the agent from startup to shutdown (not just grpc calls).

We may collectively decide that we only care about gRPC tracing. That certainly simplifies the problem significantly. However, I don't think it's ever going to be as useful as agent start-to-exit tracing as that will show what happened before the gRPC server came up for example.